state/factory/workingset.go

```diff
@@ -517,8 +545,14 @@ func updateReceiptIndex(receipts []*action.Receipt) {
 }
 
 func (ws *workingSet) ValidateBlock(ctx context.Context, blk *block.Block) error {
-	if err := ws.validateNonce(ctx, blk); err != nil {
-		return errors.Wrap(err, "failed to validate nonce")
+	if protocol.MustGetFeatureCtx(ctx).SkipSystemActionNonce {
+		if err := ws.validateNonce2(ctx, blk); err != nil {
+			return errors.Wrap(err, "failed to validate nonce")
+		}
+	} else {
+		if err := ws.validateNonce(ctx, blk); err != nil {
+			return errors.Wrap(err, "failed to validate nonce")
+		}
```

Comment:
block 1 has 2 tx, from the same sender `io1vtm2zgn830pn6auc2cvnchgwdaefa9gr4z0s86`, both with `nonce=0`, one is a transfer, the other is grant reward (system action)

`validateNonce2()` will ignore the grant reward, but return error for the transfer tx (since it's expecting nonce=1 but got nonce=0), so we have to keep the height check and 2 validate()